File: user-service/src/main/java/io/ecommerce/userservice/core/domain/dto/response/OrderResponse.java
package io.ecommerce.userservice.core.domain.dto.response;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * @author : choi-ys
 * @date : 2022/01/17 11:12 오전
 */
@Getter
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class OrderResponse {
    private Long id;
    private Long userId;
    private Long productId;
    private Integer quantity;
    private Long unitPrice;
    private Long totalPrice;
    private LocalDateTime createdAt;

    private OrderResponse(
            Long id,
            Long userId,
            Long productId,
            Integer quantity,
            Long unitPrice,
            Long totalPrice,
            LocalDateTime createdAt
    ) {
        this.id = id;
        this.userId = userId;
        this.productId = productId;
        this.quantity = quantity;
        this.unitPrice = unitPrice;
        this.totalPrice = totalPrice;
        this.createdAt = createdAt;
    }

    public static OrderResponse of(
            Long id,
            Long userId,
            Long productId,
            Integer quantity,
            Long unitPrice,
            Long totalPrice,
            LocalDateTime createdAt
    ) {
        return new OrderResponse(
                id,
                userId,
                productId,
                quantity,
                unitPrice,
                totalPrice,
                createdAt
        );
    }
}

File: order-service/src/main/kotlin/io/ecommerce/orderservice/core/domain/dto/response/OrderResponse.kt
package io.ecommerce.orderservice.core.domain.dto.response

import io.ecommerce.orderservice.core.domain.dto.entity.Orders
import java.time.LocalDateTime

/**
 * @author : choi-ys
 * @date : 2022/01/09 7:57 오후
 */
data class OrderResponse(
    val id: Long,
    val userId: Long,
    val productId: Long,
    val quantity: Int,
    val unitPrice: Long,
    val createdAt: LocalDateTime,
) {
    companion object {
        fun of(orders: Orders): OrderResponse {
            return OrderResponse(
                id = orders.id,
                userId = orders.userId,
                productId = orders.productId,
                quantity = orders.quantity,
                unitPrice = orders.unitPrice,
                createdAt = orders.createdAt.let { LocalDateTime.now() }
            )
        }
    }
}
